docs/source/user_guide.rst

```diff
@@ -0,0 +1,727 @@
+
+Retrievers all expose a `search` method that we will discuss in the next sections.
+
+See :ref:`retrievers`.
+
+Vector Retriever
+===================
+
+The simplest method to instantiate a vector retriever is:
+
+.. code:: python
+
+    from neo4j_genai.retrievers import VectorRetriever
+
+    retriever = VectorRetriever(
+        driver,
+        index_name=POSTER_INDEX_NAME,
+    )
+
+The `index_name` is the name of the Neo4j vector index that will be used for similarity search.
+
+
+.. warning::
+
+    Vector index use an **approximate nearest neighbor** algorithm.
+    Refer to the `Neo4j Documentation <https://neo4j.com/docs/cypher-manual/current/indexes/semantic-indexes/vector-indexes/#_limitiations_and_known_issues>`_ to learn about its limitations.
+
+
+Search Similar Vector
+-----------------------------
+
+To identify the top 3 most similar nodes, perform a search by vector:
+
+.. code:: python
+
+    vector = []  # a list of floats, same size as the vectors in the Neo4j vector index
+    retriever_result = retriever.search(query_vector=vector, top_k=3)
+
+However, in most cases, a text (from the user) will be provided instead of a vector.
+In this scenario, an `Embedder` is required.
+
+Search Similar Text
+-----------------------------
+
+When searching for a text, specifying how the retriever transforms (embeds) the text
+into a vector is required. Therefore, the retriever requires knowledge of an embedder:
+
+.. code:: python
+
+    embedder = OpenAIEmbeddings(model="text-embedding-3-large")
+
+    # Initialize the retriever
+    retriever = VectorRetriever(
+        driver,
+        index_name=POSTER_INDEX_NAME,
+        embedder=embedder,
+    )
+
+    query_text = "How do I do similarity search in Neo4j?"
+    retriever_result = retriever.search(query_text=query_text, top_k=3)
+
+
+Embedders
+-----------------------------
+
+Currently, this package supports two embedders: `OpenAIEmbeddings` and `SentenceTransformerEmbeddings`.
+
+The `OpenAIEmbedder` was illustrated previously. Here is how to use the `SentenceTransformerEmbeddings`:
+
+.. code:: python
+
+    from neo4j_genai.embeddings import SentenceTransformerEmbeddings
+
+    embedder = SentenceTransformerEmbeddings(model="all-MiniLM-L6-v2")  # Note: this is the default model
+
+If another embedder is desired, a custom embedder can be created. For example, consider
+the following implementation of an embedder that generates random numbers:
+
+.. warning::
+    Do not use it in your application :)
+
+
+.. code:: python
+
+    import random
+    from neo4j_genai.embedder import Embedder
+
+    class RandomEmbedder(Embedder):
+        def __init__(
+            self,
+            size: int,
+            seed: int = 42,
+        ) -> None:
+            self.size = size
+            random.seed(seed)
+
+        def embed_query(self, text: str) -> list[float]:
+            return [
+                random.random()
+                for _ in range(self.size)
```

Comment:
Should we provide an actual example rather than one that produces random vectors here? 

I'm quite fine with either way, but wdyt?

Reply:
I don't have a lot of ideas to be honest, OpenAI + SentenceTransformers are already covering a lot of use cases. But open to any suggestion :-D

Reply:
```python
from llama_index.embeddings.ollama import OllamaEmbedding
from neo4j_genai.embedder import Embedder

class OllamaEmbedder(Embedder):
    def __init__(self, ollama_embedding): 
        self.embedder = ollama_embedding
        
    def embed_query(self, text: str) -> list[float]:
        embedding = self.embedder.get_text_embedding_batch(
            [text], show_progress=True
        )
        return embedding[0] 

ollama_embedding = OllamaEmbedding(
            model_name="llama3",
            base_url="http://localhost:11434",
            ollama_additional_kwargs={"mirostat": 0},
        )
embedder = OllamaEmbedder(ollama_embedding)
vector = embedder.embed_query("some text")  
```

Perhaps one with Ollama? This example would work but I'm wondering if it's overcomplicating it

Reply:
I like it, it gives an example of wrapping LlamaIndex objects as well.

Reply:
https://github.com/neo4j/neo4j-genai-python/pull/72/commits/ba4d3231ec7820bf526cd011e2e2b97155e747bc